src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1034,13 +1047,109 @@ impl_compose_tree (const char      *treefile_pathstr,
 
   g_print ("%s => %s\n", self->ref, new_revision);
 
+  /* Finally process the --touch-if-changed option  */
   if (!process_touch_if_changed (error))
     return FALSE;
 
   return TRUE;
 }
 
 int
+rpmostree_compose_builtin_install_root (int             argc,
+                                        char          **argv,
+                                        RpmOstreeCommandInvocation *invocation,
+                                        GCancellable   *cancellable,
+                                        GError        **error)
+{
+  g_autoptr(GOptionContext) context = g_option_context_new ("TREEFILE DESTDIR");
+  if (!rpmostree_option_context_parse (context,
+                                       option_entries,
```

Comment:
We have to break the option array up into install and commit options now, right? E.g. `--add-metadata-string` doesn't make sense for `install-root`. Maybe let's do like the pkg builtins and split them up in two and in `compose_builtin_tree`, just use `g_option_context_add_main_entries` to combine them?

Reply:
I thought about that...but OTOH down the line I'd like to store state like that in a file in the root, and pick it back up.   Though anyways you're right in the short term we should do that.